aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -74,17 +74,14 @@ class S3URI {
     this.scheme = schemeSplit[0];
 
     String[] authoritySplit = schemeSplit[1].split(PATH_DELIM, 2);
-    ValidationException.check(
-        authoritySplit.length == 2, "Invalid S3 URI, cannot determine bucket: %s", location);
-    ValidationException.check(
-        !authoritySplit[1].trim().isEmpty(), "Invalid S3 URI, path is empty: %s", location);
+
```

Comment:
Never realized this class was package private. So that does give us the flexibility to change the behavior without impacting users. I think it makes sense then, AmazonS3URI in the AWS SDK has the same semantics for allowing bucket by itself as a URI. I do think we should confirm with @danielcweeks on if this was intentionally implemented this way in Iceberg.

Reply:
@danielcweeks Could you check this comment?

Reply:
Thanks for reviewing, Daniel. 
> If you use the bucket as the location of a table, you basically cannot use that bucket for anything else. Things like orphan files procedure will wipe anything else in the bucket out.
> 
> Is there a requirement that you cannot have a bucket prefix for some use case?

Yes, I agree this, but I understand there are users who build their Iceberg tables on each S3 bucket with some configurations such as encryptions, bucket policy, lifecycle policy etc. Currently, Iceberg doesn’t allow users to directly create an Iceberg table at S3 bucket, but I think it’s possible to provide creating an Iceberg table with the users.

Let me share more background for this PR. This PR is based on the following situation I confirmed:
1. You initially build an Iceberg table at direct S3 bucket like `s3://bucket` using Athena or Spark (we can also build an Iceberg table at direct S3 bucket with `LOCATION` clause in `CREATE TABLE` DDL)
2. Then, you write data into the Iceberg table by write operation such as `INSERT INTO`. This is successful.
3. If we run a read query (e.g. `SELECT`) for the Iceberg table after writing data, it fails with `org.apache.iceberg.exceptions.ValidationException : Invalid S3 URI, path is empty: s3://bucket`.

Users who have this situation need to re-create the Iceberg table at s3 bucket with path(s) such as `s3://bucket/path`. I confirmed there are users who had this situation.

To avoid this situation, we can have three choices (in my understanding):
* Just showing warning message to users if they directly create an Iceberg table at s3 bucket
* Not allowing users to directly create an Iceberg table at s3 bucket
* Allowing users to directly create an Iceberg table at s3 bucket

Initially I thought adding the restriction of table creation in regards to the first two choices. However based on the following considerations, I choose the last (third) one and decided to submit this PR.
* For the first one, it’s not difficult because just putting warning message in the Iceberg code. But this sometimes wouldn’t work well for users to avoid creating an Iceberg table at direct s3 bucket.
* The second one needs to add the restriction of table creation at direct s3 bucket when users specify a s3 bucket name only for an Iceberg table location. However, this restriction forces users to use s3 bucket with paths and it isn’t consistent with other software behavior that allows to use direct s3 bucket.

Sorry for my long comments, if there’s my misunderstandings, please let me know.

Reply:
Thanks for the context.  I didn't realize this was possible via Athena or a pattern anyone would use.  I feel like we wouldn't recommend using a bucket as the primary location, but since there is precedent and it would make it easier to convert to Iceberg, supporting this makes sense.